src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/PileupDetectionArgumentCollection.java

```diff
@@ -0,0 +1,92 @@
+    public static final String PILEUP_DETECTION_SNP_THRESHOLD = "pileup-detection-enable-snp-alt-threshold";
+    public static final String PILEUP_DETECTION_ABSOLUTE_ALT_DEPTH = "pileup-detection-absolute-alt-depth";
+    public static final String PILEUP_DETECTION_INDEL_THRESHOLD = "pileup-detection-enable-indel-alt-threshold";
+    public static final String PILEUP_DETECTION_FILTER_COVERAGE_LONG_NAME = "pileup-detection-filter-coverage-threshold";
+
+    //TODO we currently don't see the same threshold from active region determination...
+    public static final String PILEUP_DETECTION_ACETIVE_REGION_LOD_THRESHOLD_LONG_NAME = "pileup-detection-active-region-lod-threshold";
+
+
+    // Arguments related to DRAGEN heuristics related to "read badness" intended to filter out false positives from the pileup deteciotn code
+    public static final String PILEUP_DETECTION_BAD_READ_RATIO_LONG_NAME = "pileup-detection-bad-read-tolerance";
+    public static final String PILEUP_DETECTION_PROPER_PAIR_READ_BADNESS_LONG_NAME = "pileup-detection-proper-pair-read-badness";
+    public static final String PILEUP_DETECTION_EDIT_DISTANCE_BADNESS_LONG_NAME = "pileup-detection-edit-distance-read-badness-threshold";
+    public static final String PILEUP_DETECTION_CHIMERIC_READ_BADNESS_LONG_NAME = "pileup-detection-chimeric-read-badness";
+    //TODO these need to be implemented with some input from Illumina
+    public static final String PILEUP_DETECTION_TLEN_MEAN_LONG_NAME = "pileup-detection-template-mean-badness-threshold";
+    public static final String PILEUP_DETECTION_TLEN_STD_LONG_NAME = "pileup-detection-template-std-badness-threshold";
+
+    // Argumetns related to filtering the list at the assembly graph level
+    public static final String PILEUP_DETECTION_INDEL_SNP_BLOCKING_RANGE = "pileup-detection-snp-adjacent-to-assembled-indel-range";
+
+    /**
+     * Enables pileup-based haplotype creation and variant detection
+     *
+     * NOTE: --pileup-detection is a beta feature. Use this mode at your own risk.
+     */
+    @Argument(fullName= PILEUP_DETECTION_LONG_NAME, doc = "If enabled, the variant caller will create pileup-based haplotypes in addition to the assembly-based haplotype generation.", optional = true)
+    public boolean usePileupDetection = false;
+
+    /**
+     * Enables detection of indels from the pileups in. (EXPERIMENTAL FEATURE)
+     */
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_ENABLE_INDELS, doc = "If enabled, pileup detection code will attempt to detect indels missing from assembly. (Requires `--pileup-detection` argument)", optional = true)
+    public boolean detectIndels = false;
+
+
+    /**
+     * Percentage of reads required to supprot the alt for a variant to be considered
+     */
+    @Advanced
+    @Argument(fullName= PILEUP_DETECTION_SNP_THRESHOLD, doc = "Percentage of alt supporting reads in order to consider alt SNP", optional = true)
```

Comment:
I would rename argument PILEUP_SNP_DETECTION_RATIO (also below for INDEL)

Reply:
It would be good to note in the `doc=` of all these arguments  that it pertains to the PILEUP detection only. otherwise the only hint to that it in the name of the argument itself.

Reply:
I've changed all of the arguments to have a "Pileup Detection: " argument. Hopefully that will make it a little bit clearer? I've hidden these arguments anyway so maybe its all moot...